File: android/app/src/main/kotlin/com/example/flutter_example_app/StrutFitTrackingManager.kt
package com.example.flutter_example_app

import android.app.Activity
import android.util.Log
import io.flutter.embedding.engine.FlutterEngine
import io.flutter.plugin.common.MethodChannel
import strutfit.button.models.ConversionItem
import strutfit.button.StrutFitTracking

class StrutFitTrackingManager(
    activity: Activity,
    flutterEngine: FlutterEngine
) {

    private val channelName = "STRUTFIT_TRACKING_PIXEL_CHANNEL"
    private val trackingHandler = StrutFitTrackingHandler(activity)

    init {
        registerMethodChannel(flutterEngine)
    }

    private fun registerMethodChannel(flutterEngine: FlutterEngine) {
        MethodChannel(flutterEngine.dartExecutor.binaryMessenger, channelName).setMethodCallHandler { call, result ->
            if (call.method == "registerOrderForStrutFit") {
                val args = call.arguments as? Map<String, Any>
                if (args != null) {
                    trackingHandler.registerOrderForStrutFit(args, result)
                } else {
                    result.error("INVALID_ARGS", "Arguments were null", null)
                }
            } else {
                result.notImplemented()
            }
        }

        Log.d("StrutFitTrackingManager", "✅ Tracking pixel channel registered successfully")
    }

    class StrutFitTrackingHandler(private val activity: Activity) {

        fun registerOrderForStrutFit(args: Map<String, Any>, result: MethodChannel.Result) {
            try {
                val organizationId = args["organizationId"] as Int
                val orderReference = args["orderReference"] as String
                val orderValue = args["orderValue"] as Double
                val currencyCode = args["currencyCode"] as String
                val userEmail = args["userEmail"] as String?

                val strutFitTracking = StrutFitTracking(activity, organizationId)
                val items = ArrayList<ConversionItem>()

                val productItems = args["items"] as List<Map<String, Any>>
                for (item in productItems) {
                    val productIdentifier = item["productIdentifier"] as String
                    val price = item["price"] as Double
                    val quantity = item["quantity"] as Int
                    val size = item["size"] as String
                    val conversionItem = ConversionItem(productIdentifier, price, quantity, size)
                    items.add(conversionItem)
                }

                strutFitTracking.registerOrder(orderReference, orderValue, currencyCode, items, userEmail)
                Log.d("StrutFitTracking", "✅ Order Tracked: $orderReference")
                result.success("Order Tracked Successfully")
            } catch (e: Exception) {
                Log.e("StrutFitTracking", "❌ Error tracking order: ${e.message}", e)
                result.error("TRACKING_ERROR", e.message, null)
            }
        }
    }
}

File: ios/Runner/StrutFitTrackingManager.swift
import Foundation
import Flutter
import StrutFitButtonSDK

class StrutFitTrackingManager {
    
    // MARK: - Properties
    private weak var controller: FlutterViewController?
    private let trackingChannel = "STRUTFIT_TRACKING_PIXEL_CHANNEL"
    private var trackingHandler: StrutFitTrackingHandler!
    
    // MARK: - Initializer
    init(controller: FlutterViewController) {
        self.controller = controller
        setupTrackingHandler()
        setupTrackingPixelChannel()
    }
    
    /// Initializes the tracking handler.
    private func setupTrackingHandler() {
        trackingHandler = StrutFitTrackingHandler()
    }
    
    /// Sets up a MethodChannel for tracking pixel requests from Flutter.
    private func setupTrackingPixelChannel() {
        guard let controller = controller else {
            print("❌ No FlutterViewController available for tracking channel")
            return
        }
        
        let channel = FlutterMethodChannel(name: trackingChannel, binaryMessenger: controller.binaryMessenger)
        
        channel.setMethodCallHandler { [weak self] (call, result) in
            guard let self = self else { return }
            
            switch call.method {
            case "registerOrderForStrutFit":
                self.handleRegisterOrderForStrutFit(call: call, result: result)
                
            default:
                result(FlutterMethodNotImplemented)
            }
        }
        
        print("✅ Tracking pixel channel registered successfully")
    }
    
    /// Handles the "registerOrderForStrutFit" method call.
    private func handleRegisterOrderForStrutFit(call: FlutterMethodCall, result: @escaping FlutterResult) {
        guard let args = call.arguments as? [String: Any] else {
            print("❌ Invalid arguments received for registerOrderForStrutFit")
            result(FlutterError(code: "INVALID_ARGS", message: "Arguments were null", details: nil))
            return
        }
        
        print("📦 Received order registration request: \(args)")
        trackingHandler.registerOrderForStrutFit(args: args, result: result)
    }
}

class StrutFitTrackingHandler {
    
    func registerOrderForStrutFit(args: [String: Any], result: @escaping FlutterResult) {
        do {
            guard
                let organizationId = args["organizationId"] as? Int,
                let orderReference = args["orderReference"] as? String,
                let orderValue = args["orderValue"] as? Double,
                let currencyCode = args["currencyCode"] as? String,
                let userEmail = args["userEmail"] as? String,
                let itemsArray = args["items"] as? [[String: Any]]
            else {
                result(FlutterError(code: "INVALID_ARGS", message: "Missing required arguments", details: nil))
                return
            }
            
            let tracking = StrutFitTracking(organizationUnitId: organizationId)
            let items = itemsArray.map {
                ConversionItem(
                    productIdentifier: $0["productIdentifier"] as? String ?? "",
                    price: Float($0["price"] as? Double ?? 0),
                    quantity: $0["quantity"] as? Int ?? 0,
                    size: $0["size"] as? String ?? ""
                )
            }

            tracking.registerOrder(orderReference: orderReference, orderValue: Float(orderValue), currencyCode: currencyCode, items: items, userEmail: userEmail)
            result("Tracking success!")
        } catch {
            result(FlutterError(code: "TRACKING_ERROR", message: "Failed to register", details: error.localizedDescription))
        }
    }
}
